### PR TITLE
Add support for containerised azure packer builds

### PR DESCRIPTION
**Pull request by bpradipt**

### CODE REVIEW

**Comment by bpradipt** [comment]
Tested by building ubuntu and centos images

**Review by snir911 — COMMENTED** [review]
add minor comments, overall LGTM
BTW, have you tried RHEL?

File: azure/image/Dockerfile
Comment (bpradipt) [reply azure/image/Dockerfile:7]: I could run it as-is without requiring to install anything additional

File: azure/image/misc-settings.sh
Comment (bpradipt) [reply azure/image/misc-settings.sh:18]: Right. I was thinking about having the same misc_script for all providers and then switching to a common script in a subsequent PRs. Wdyt ?

**Comment by bpradipt** [comment]
> add minor comments, overall LGTM BTW, have you tried RHEL?
I'm yet to try. I tried only CentOS and Ubuntu. Although I suspect RHEL will be very similar to CentOS. I'll try and update here.

File: azure/image/misc-settings.sh
Comment (snir911) [reply azure/image/misc-settings.sh:18]: IIRC i tried to point to the file under podvm, i don't recall why i didn't eventually, go ahead, I'll test

Comment (bpradipt) [reply azure/image/misc-settings.sh:18]: If you also agree for a common file, then in this series can I go ahead with the copy and then subsequently we can change for aws, azure to point to files from podvm. I think some directory adjustments will be needed which will require testing the workflows etc.

**Comment by bpradipt** [comment]
@snir911 Added support for rhel and tried a build successfully for rhel as well.

File: azure/README.md
Comment (snir911) [inline azure/README.md:83]: can it be done from within the dockerfile?

Comment (bpradipt) [reply azure/README.md:83]: Technically it can be done, but it's better to keep it independent of the Dockerfile for the following reasons:
1. Requires az cli to be available in the container image
2. Requires authentication to be performed inside the container
3. Not every image requires license acceptance
4. The acceptance is global for the subscription. So preferred base images should already be available and attached to the subscription.

File: azure/image/Dockerfile
Comment (snir911) [inline azure/image/Dockerfile:58]: AFAIU passing credentials to a docker build is bad practice, maybe worth to add a disclaimer in the doc

**Review by snir911 — APPROVED** [review]
Overall LGTM, see comments, also i'd squash some of the commits, not very important though..

**Comment by bpradipt** [comment]
/hold
Investigating usage of `docker build --secret ...`

**Comment by bpradipt** [comment]
I have now switched to using `docker build --secret ..`
example usagE:
```
export MYSECRET=mypass
docker build --secret id=MYSECRET -t my_image -f Dockerfile .
```
However podman doesn't seem to work with using env variables as secret.

**Review by snir911 — APPROVED** [review]
LGTM, thanks!
FWIW it should b possible to use with podman by installing buildkit in separate, however, it doesn't seem very straight forward as in docker..

File: azure/image/Makefile
Comment (kartikjoshi21) [inline azure/image/Makefile:1]: Please look at this change https://github.com/confidential-containers/cloud-api-adaptor/pull/866. @surajssd have changes to modify how we build image via packer.

**Review by kartikjoshi21 — COMMENTED** [review]
Just need to be modified as per new changes in https://github.com/confidential-containers/cloud-api-adaptor/pull/866. Otherwise LGTM

**Review by kartikjoshi21 — APPROVED** [review]
LGTM, https://github.com/confidential-containers/cloud-api-adaptor/issues/865 this one can be done after this is merged.

**Pull request merged**
